### ISSUE TITLE
Muon optimizer implementation mismatch with reference and MoonshotAI versions

### ISSUE THREAD

**Issue by minpeter** [kozistr/pytorch_optimizer#408]
**This issue report was generated by AI, so it may contain inaccurate information. However, the experimental results below are actual test results, and this is why I'm leaving this issue.**

---

## Describe the bug
The `Muon` optimizer in `pytorch_optimizer` exhibits significantly different training dynamics compared to both the official KellerJordan/Muon implementation and the MoonshotAI/Moonlight reference. Specifically, the training loss and gradient norm do not decrease as expected, indicating a potential mis-implementation of the Newton–Schulz orthogonalization or update scaling.

## To Reproduce
* OS : Linux (Ubuntu 22.04)
* PyTorch version : 2.0.1
* Python version : 3.10
* pytorch-optimizer version : 3.3.0
* reproducible codes :
```python
import torch
from pytorch_optimizer import Muon

# Simple model
model = torch.nn.Linear(20, 10)
optimizer = Muon(model.parameters(), lr=1e-3, ns_steps=5)

for step in range(500):
    optimizer.zero_grad()
    x = torch.randn(8, 20)
    y = model(x)
    loss = y.sum()
    loss.backward()
    optimizer.step()
```

## Log
<img width="483" height="582" alt="Image" src="https://github.com/user-attachments/assets/42ef8c65-7a51-44e1-8382-abfe9fc6a237" />

## Expected behavior
The loss and gradient norms using `pytorch_optimizer.Muon` should closely mirror those from:
- Official reference: https://github.com/KellerJordan/Muon
- MoonshotAI reference: https://github.com/MoonshotAI/Moonlight

## Additional context
Discrepancy likely originates from one or more of:
- Coefficients `(a, b, c)` for the quintic Newton–Schulz iteration  
- The order and dtype of spectral norm normalization  
- Placement of weight decay relative to the orthogonalization step

**Comment by kozistr** [comment]
@minpeter Hi! Thanks for raising this issue!

I've tested on my local environment with three different implementations of Muon optimizer (pytorch-optimizer 3.3.0, 3.7.0 (latest), KellerJordan/Muon (latest)), but I'm unable to reproduce the issue with your example.

Could you please share the details of how you use Muon optimizer and test with the latest `pytorch-optimizer`?

Here's my W&B log below.

<img width="1427" height="411" alt="Image" src="https://github.com/user-attachments/assets/8a2ab85c-efc2-4b1d-b253-50f3410afd15" />

And code

```Python
import torch

import wandb
import lightning.pytorch as pl
from pytorch_optimizer import Muon

pl.seed_everything(42)

# creating Muon optimizer with `pytorch-optimizer`
model = torch.nn.Linear(20, 10)

muon_params = [p for p in model.parameters() if p.ndim >= 2]
non_muon_params = [p for p in model.parameters() if p.ndim < 2]

optimizer = Muon(muon_params, lr=1e-3, ns_steps=5, adamw_params=non_muon_params)

for step in range(500):
    optimizer.zero_grad()
    x = torch.randn(8, 20)
    y = model(x)
    loss = y.sum()
    loss.backward()
    optimizer.step()

    wandb.log({'train/loss': loss, 'train/grad_norm': torch.norm(model.weight).item()})

...

# creating Muon optimizer with KellerJordan/Muon implementation
model = torch.nn.Linear(20, 10)

muon_params = [p for p in model.parameters() if p.ndim >= 2]
non_muon_params = [p for p in model.parameters() if p.ndim < 2]

param_groups = [
    dict(params=muon_params, use_muon=True, lr=1e-3, weight_decay=0.01),
    dict(params=non_muon_params, use_muon=False, lr=3e-4, betas=(0.9, 0.95), weight_decay=0.0),
]
optimizer = SingleDeviceMuonWithAuxAdam(param_groups)

...
```

Besides the above, I've updated the implementation here when I found the update in the official implementation. So, there may be a gap among `pytorch-optimizer` versions due to differences in default hyperparameters (e.g. learning rate, betas) and some changes.

And, I can confirm that `pytorch-optimizer 3.3.0 Muon` is identical to the initial version of Muon, and `3.7.0 Muon` is similar to the latest version of Muon, `MuonWithAuxAdam` more precisely.

Feel free to leave any comments or feedback :)

Thanks!

**Comment by minpeter** [comment]
You can also use this to reproduce it: https://github.com/minpeter/krill
